.github/workflows/workflow-merge-to-main.yaml

```diff
@@ -0,0 +1,50 @@
+# This workflow is called when a pull request  is approved
+# that targets the protected branch `main`.
+# The job `merge-to-main` ("Status check for enabling merging to main")
+# is a required status check.
+# (https://docs.github.com/en/rest/commits/statuses?apiVersion=2022-11-28)
+# We can only merge to main when this job has run and is green.
```

Comment:
I guess technically, there is a chance that once approved and checks have passed but BEFORE merging - something can get merged into main that is not compatible with this branch causing issues.

Though would assume once you approve something the intent is to also merge it quite fast.

Reply:
Would it make sense to run the checks whenever there is a merge to main? 🤔

Reply:
>I guess technically, there is a chance that once approved and checks have passed but BEFORE merging

That should be prevented because a protected branch rules would be setup up with `Require branches to be up to date before merging`.  I tested that scenario and it did prevent me from merging. (still would like to see it actually work in real-life because I had my `fake-main` etc.)

> Would it make sense to run the checks (in addition) whenever there is a merge to main? 🤔

We can do that additionally if we want to be super safe, yes! That maybe a good idea. We even could (well, in the future) add checks there like running code coverage etc. and have the nice little badged that other repositories sometimes have.